RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+  userPoolId: "us-east-1_0yqxxHm5q",
+  userPoolClientId: "3keodiqtm52nhh2ls0vQfs5v1q",
+  signUpVerificationMethod: "code"
+};
+
+Amplify.configure({
+  Auth: authConfig
+});
+```
+**Proposed Amplify v6 DX**
+
+https://user-images.githubusercontent.com/70438514/222543618-d5c42798-28c5-41e9-9ff6-c9061b86ef43.mov
+
+<br />
+
+Try out the new types here: https://www.typescriptlang.org/play#gist/7332e86effa3b0d627eebfc71fad1aca
+
+# `Auth` Category Changes
+
+Amplify is proposing the following changes for the `Auth` category. Similar changes will be applied across all of the `Auth` APIs but examples for specific APIs are highlighted below.
+
+## TypeScript support for user attributes
+
+User attributes inference on the `signUp` API.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+
+Auth.signUp({
+  username: "username",
+  password: "*******",
+  attributes: {
+    email: "email@domain.com"
+  }
+});
+
+```
+
+**Amplify v5 DX**
+
+![signup-v5](https://user-images.githubusercontent.com/70438514/222456624-b4af7349-db9d-4304-9cdc-08cf649e2d31.png)
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+
+signUp({
+  username: "username",
+  password: "********",
+  options: {
+    userAttributes: [{ userAttributeKey: "email", value: "email@domain.com" }],
+  },
+});
+
+```
+
+**Proposed Amplify v6 DX**
+
+<img width="556" alt="signup-v6" src="https://user-images.githubusercontent.com/70438514/222456627-f196eedf-90e5-4bb0-81a7-78e11d88b09b.png">
+
+## Predictable API responses
+
+We are improving **_DX_** by providing descriptive API responses to help customers complete auth flows. An example for the `confirmSignUp` API is highlighted below.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+
+const resp = await Auth.confirmSignUp("username", "112233")
+
+if (resp === "SUCCESS"){
+  // Show login component
+}
+
+```
+
+**Amplify v5 DX**
+
+![confirmSignUp-v5](https://user-images.githubusercontent.com/70438514/222456621-fca2d313-3d39-4a1e-8bcb-a1e86235bd06.png)
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+
+const resp = await confirmSignUp({
+  username: "username",
+  confirmationCode: "112233",
+});
+
+if (resp.isSignUpComplete) {
+  // Show login component
+}
+
+```
+
+**Proposed Amplify v6 DX**
+
+<img width="594" alt="confirmSignUp-v6" src="https://user-images.githubusercontent.com/70438514/222456623-34d48988-5e3d-43be-bc9a-7c6d8ecb0b00.png">
+
+Try out the new types here: https://stackblitz.com/edit/rfc-auth?file=auth.ts
+
+# `Storage` Category Changes
+
+Amplify is proposing the following changes for the `Storage` category.
+
+## Introduction of object reference types
+
+In order to permit better inoperability between `storage` APIs and to enable future improvements such as more granular bucket management, we will introduce `StorageObjectReference` & `StoragePrefixReference` types to represent items in cloud storage. An example for copying an object from one access level to another is highlighted below.
```

Comment:
"interoperability"

Reply:
> and to enable future improvements such as more granular bucket management

I would not promise this on this RFC, just to not create expectations for V6 about this topic

Reply:
at some point we will solve this but dont have a timeline for that

Reply:
Good points both of you